scripts/globals/spells/white/silencega.lua

```diff
@@ -41,6 +41,8 @@ spellObject.onSpellCast = function(caster, target, spell)
 
         if target:addStatusEffect(effectType, 1, 0, resduration) then
             spell:setMsg(xi.msg.basic.MAGIC_ENFEEB_IS)
+            -- only increment the resbuild if successful (not on a no effect)
+            xi.magic.incrementBuildDuration(target, params.effect, caster)
```

Comment:
can we do this `OnEffectGain` instead so we avoid calling it for every spell?

Reply:
> can we do this `OnEffectGain` instead so we avoid calling it for every spell?

I could but I do not think this would decrease the number of times such functions are actually called as in the spell script it also gets called only if the spell actually puts on the status effect. Having it in the status effect script (rather than the spell) would mean slightly fewer lines of code as some spells cause the same status effect, but at the cost of losing some understandability as the two related functions of calculateBuildDuration and incrementBuildDuration would no longer be near each other.